### PR TITLE
chore(pg proxy): update cloud cert

### PR DESCRIPTION
**Pull request by greyscaled**
This removes extraneous cert generation for pg proxy. The test was updated to use certs with the ED25519 algorithm. I didn't feel it was important enough to source control cert generation, since there's no need to rotate these in tests. For posterity, I used:

```shell
openssl genpkey -out key.pem -algorithm ED25519
openssl x509 -new -key key.pem -subj "/C=US/ST=Texas/L=Austin/O=GlareDB/OU=Engineering/CN=glaredb.com"
```

Resolves: #2270

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,23 +4,16 @@ FROM rust:1.73-bookworm AS builder
 WORKDIR /usr/src/glaredb
 COPY . .
 
-RUN apt-get update && apt-get install -y openssl ca-certificates
```
Comment (greyscaled) [inline Dockerfile:7]: (review): I'm expecting that we only needed these to run the openssl x509 gen. If a build dependency relies on `ca-certificates`, I'd want to know why (and to ensure it's documented).

File: Dockerfile [hunk 2]
```diff
@@ -4,23 +4,16 @@ FROM rust:1.73-bookworm AS builder
...
 # Build release binary.
 RUN just build --release
 
-# Generate certs.
-RUN ./scripts/gen-certs.sh
-
 FROM debian:bookworm-slim
 
 # Runtime deps.
 RUN apt-get update -y && apt-get install -y openssl ca-certificates openssh-client
```
Comment (greyscaled) [inline Dockerfile:14]: I believe our tunnels feature relies on these.

File: Dockerfile (hunk 1 above)
Comment (scsmithr) [reply Dockerfile:7]: If it builds without it, then probably fine. I could see trying to fetch the protoc binary potentially fail without ca certs.

File: Dockerfile (hunk 2 above)
Comment (scsmithr) [reply Dockerfile:14]: Yup, we'll want to keep these in.

File: Dockerfile (hunk 1 above)
Comment (greyscaled) [reply Dockerfile:7]: CI build step passed, and I succesfully `docker build .` locally.

File: crates/pgsrv/src/ssl.rs [hunk 3]
```diff
@@ -148,54 +148,21 @@ mod tests {
...
-klokrmYS4DhlIHrSnJTfpPymPhqsSVHd97Jr20nbuBbXlRsCAwEAAaNTMFEwHQYD
-VR0OBBYEFHMbgh0q1cxHVpR0UCWq92+0zQrsMB8GA1UdIwQYMBaAFHMbgh0q1cxH
-VpR0UCWq92+0zQrsMA8GA1UdEwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggEB
-AEJ554uXpkgXLGECVueikjzGOyCb7aBmKYdssuutWpEGM3pZpyCKzz/tsanULH3Q
-kNu5/eJYc0BBI2YXz1eIFrJR3fM+qbpIvP1j67CLhsjtvKSQTFEvqseCCZ1l34WK
-dNzpwf0HMB5MAt6T8xeQ5gDRNo7C3HuKT/8WouaIdn9H2UEluQBDM5Yg+cpDWOlv
-Dr/IzcLC/pyEmx7spkPfvxebtY64kdFh8qyl9iWZLO9wrpaWY1vV41sjZ1F41AgY
-KaRWGJhHwhyDcYzelsn0Ew2LdB7WfFQZfZe9EHJTj7bd6y6AbkduZ31oebXYlOzo
-1DkrU2K5aMlVF/vExWpfChk=
+MIIBkzCCAUUCFGipMcv8Oq6O89V+OkbybaF4q3XaMAUGAytlcDBsMQswCQYDVQQG
```
Comment (tychoish) [inline crates/pgsrv/src/ssl.rs:151]: what kind of certificate is this? and where did it come from (I know it's in a test)

**Review by tychoish — APPROVED** [review]

File: crates/pgsrv/src/ssl.rs (hunk 3 above)
Comment (tychoish) [reply crates/pgsrv/src/ssl.rs:151]: oh I see.

**Pull request merged**